test/infrastructure/docker/e2e/docker_test.go

```diff
@@ -179,6 +180,36 @@ var _ = Describe("Docker", func() {
 					},
 				}
 				framework.AssertControlPlaneFailureDomains(ctx, assertControlPlaneFailureDomainInput)
+			})
+
+			Specify("upgrades etcd", func() {
+				patchHelper, err := patch.NewHelper(controlPlane, mgmtClient)
+				Expect(err).ToNot(HaveOccurred())
+				// This is the etcd version meant for k8s 1.17.x
+				// k8s 1.16.x clusters ususally get deployed with etcd 3.3.x
+				By("patching KubeadmConfigSpec etcd image tag")
+				controlPlane.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd = v1beta1.Etcd{
+					Local: &v1beta1.LocalEtcd{
+						ImageMeta: v1beta1.ImageMeta{
+							ImageTag: "3.4.3-0",
+						},
+					},
+				}
+				Expect(patchHelper.Patch(ctx, controlPlane)).To(Succeed())
+				// Get the etcd pods from the workload cluster and then verify
+				// the image of those pods are correct.
+				workloadClient, err := mgmt.GetWorkloadClient(ctx, cluster.Namespace, cluster.Name)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("waiting for etcd pods to have the expected image tag")
+				waitForPodConditionInput := framework.WaitForPodConditionInput{
+					Lister: workloadClient,
+					ListOptions: []client.ListOption{
+						client.MatchingLabels(map[string]string{"component": "etcd"}),
+					},
+					Condition: framework.EtcdImageTagCondition("3.4.3-0"),
+				}
+				framework.WaitForPodCondition(ctx, waitForPodConditionInput, "10m", "30s")
```

Comment:
Shouldn't this be part of the "Full upgrade" step, changing the imageMeta won't actually trigger an upgrade

Reply:
I was having trouble getting the `Full Upgrade` test passing on my machine. So in the interest of making progress I split it out into its own test. I'll be refactoring it further and will get it all under one test. 

I'm also going to move the upgrade tests into a separate file.